File: src/442.find_all_duplicates_in_an_array.kt
fun main() {
    println(findDuplicates(intArrayOf(4, 3, 2, 7, 8, 2, 3, 1)))
    println(findDuplicates(intArrayOf(1, 1, 2)))
    println(findDuplicates(intArrayOf(1)))
}

fun findDuplicates(nums: IntArray): List<Int> {
    val set = mutableSetOf<Int>()

    return nums.filter { !set.add(it) }
}

File: lib/442.find_all_duplicates_in_an_array.dart
void main() {
  print(findDuplicates([4, 3, 2, 7, 8, 2, 3, 1]));
  print(findDuplicates([1, 1, 2]));
  print(findDuplicates([1]));
}

List<int> findDuplicates(List<int> nums) {
  final set = Set<int>();

  return nums.where((e) => !set.add(e)).toList();
}
